swap/protocol.go

```diff
@@ -157,6 +157,8 @@ func (s *Swap) getPeer(id enode.ID) *Peer {
 type swapAPI interface {
 	Balance(peer enode.ID) (int64, error)
 	Balances() (map[enode.ID]int64, error)
+	Cheques() (map[enode.ID]map[string]*Cheque, error)
```

Comment:
I find all these API's to related to cheques somewhat confusing now.
We have 
```golang
Cheques
// Cheques returns all known last sent and received cheques, grouped by peer
PeerCheques
// PeerCheques returns the last sent and received cheques for a given peer
SentCheque
// SentCheques returns the last sent cheques for all known SWAP peers
sentCheques
// SentCheques returns the last sent cheques for all known SWAP peers
ReceivedCheques
// ReceivedCheques returns the last received cheques for all known SWAP peers
```
In my opinion, Swarm should list the APIs which are minimally needed to get sufficient information and then it will be up to other processes to process these.
This means that we can either have one call returning all known cheques, or a calls to return the last sent/received cheque *per peer*. The granularity which we have right now adds, in my opinion, a lot of redundant code with possibilities for mistakes and possibly confusion from application developers.
Please consider this comment as an open discussion point, as I am not aware of how this is done in other projects/packages and what the best strategy is here.

Reply:
it's a fair point.

if we want to keep it at a minimum, i would go with `Cheques` and `PeerCheques`. in any case, one could just use a subset of that info.

i'm not a node operator, but those seem like the more generic uses cases in my opinion. 

i can't really think of a use for _just_ the sent cheques (same for received ones). we just started out implementing those.

Reply:
addressed

Reply:
In theory, `PeerCheques` is already a subset of `Cheques`, so not essential. Having `PeerCheques` as a separate function is probably only needed when there is a single node that has a non-trivial amount of peers he ever connected with (say 1000+). I think for the sake of keeping the codebase small, we can even omit `Peercheques` and add it back later when node operators require it. 
If you think it is better to keep this call, I will be oke with that.
If we agree on removing all the other calls, let's do so! It will clean up :)

Reply:
thanks @Eknir.

as you can verify in the implementation, `PeerCheques` and `Cheques` are separate functions because one combs through the store, while the other tries to directly access a known value. so the implementations are different and it's OK that they are separate (as well as their tests). of course the peer cheques are (logically) a subset of all cheques.

i've removed the rest of the calls :+1:

Reply:
also, consider that `PeerCheques` is a very taxing function, specially after the node racks up a lot of (past) swap peers. i hope we can avoid having to call it all the time.